lib/functions/OverloadTree.ts

```diff
@@ -1,18 +1,147 @@
 import type * as E from '../expressions';
+import type { LiteralTypes } from '../util/Consts';
+import { TypeAlias, TypeURL } from '../util/Consts';
 import type { ArgumentType } from './Core';
 
 export type SearchStack = OverloadTree[];
+export type OverRideType = LiteralTypes | 'term';
+
+/**
+ * Some of the types here have some super relation.
+ * This should match the relations provided in @see{transformLiteral}.
+ * Types that are not mentioned just map to 'term'.
+ * A DAG will be created based on this. Make sure it doesn't have any cycles!
+ */
+export const extensionTableInput: Record<LiteralTypes, OverRideType> = {
+  [TypeURL.XSD_DATE_TIME]: 'term',
+  [TypeURL.XSD_BOOLEAN]: 'term',
+  [TypeURL.XSD_DATE]: 'term',
+  [TypeURL.XSD_DURATION]: 'term',
+  [TypeAlias.SPARQL_NUMERIC]: 'term',
+  [TypeAlias.SPARQL_STRINGLY]: 'term',
+  [TypeAlias.SPARQL_OTHER]: 'term',
+  [TypeAlias.SPARQL_NON_LEXICAL]: 'term',
+
+  // Datetime types
+  [TypeURL.XSD_DATE_TIME_STAMP]: TypeURL.XSD_DATE_TIME,
+
+  // Duration types
+  [TypeURL.XSD_DAYTIME_DURATION]: TypeURL.XSD_DURATION,
+  [TypeURL.XSD_YEAR_MONTH_DURATION]: TypeURL.XSD_DURATION,
+
+  // Stringly types
+  [TypeURL.RDF_LANG_STRING]: TypeAlias.SPARQL_STRINGLY,
+  [TypeURL.XSD_STRING]: TypeAlias.SPARQL_STRINGLY,
+
+  // XSD_ANY_URI is a weird one, it was handled equal to XSD_STRING before, we keep this? TODO - need to edit chart
```

Comment:
I think that's correct indeed.

Reply:
Yes, after reading https://github.com/comunica/sparqlee/issues/102#issuecomment-896876623 and in particular https://www.w3.org/TR/xpath-31/#promotion it seems like that is the right behavior. I will add the reference to the code.

Reply:
Taking this into account I think we should also allow the numeric types in this PR. https://github.com/comunica/sparqlee/issues/102#issuecomment-897103472

Reply:
Seems like it is not correct. Please take a look at https://github.com/comunica/sparqlee/issues/102#issuecomment-897427439 .

Reply:
This conversation has been resolved by type promotion.